consensus/core/src/linearizer.rs

```diff
@@ -295,6 +298,41 @@ impl Linearizer {
 
         committed_sub_dags
     }
+
+    // Try to measure the number of blocks that get pruned due to GC. This is not very accurate, but it can give us a good enough idea.
+    // We consider a block as pruned when it is an ancestor of a block that has been committed as part of the provided `sub_dag`, but
+    // it has not been committed as part of previous commits. Right now we measure this via checking that highest committed round for the authority
+    // as we don't an efficient look up functionality to check if a block has been committed or not.
+    fn update_blocks_pruned_metric(&self, sub_dag: &CommittedSubDag) {
```

Comment:
How about implementing this inside `DagState::add_commit()` before updating `last_commit`? We can check if there are blocks between last_committed_rounds per authority before updating them with the new commit, and the gc round that would result from the new commit (commit leader round - gc depth).

Reply:
Another idea is to insert gc'ed ancestors into a map here in `linearize_sub_dag()`:
```
.filter(|ancestor| {
    // Keep the block if GC is not enabled or it is enabled and the block is above the gc_round. We do this
    // to stop the recursion early and avoid going to deep when it's unnecessary.
    !gc_enabled || ancestor.round > gc_round
})
```